tools/ClassGen/InstrGen.cpp

```diff
@@ -829,6 +829,37 @@ int main(int argc, char **argv) {
       .autoVerify(VerifyKind::SameShape, {"Result", "Input"})
       .autoIRGen();
 
+  //===--------------------------------------------------------------------===//
+  //                Pre Processing
+  //===--------------------------------------------------------------------===//
+
+  BB.newInstr("AudioSpectrogram")
+      .addOperand("Spectrogram", OperandKind::Out)
+      .addOperand("Input", OperandKind::In)
+      .addOperand("Window", OperandKind::In)
+      .addOperand("TwiddleFactors", OperandKind::In)
+      .addOperand("BitReverseIndices", OperandKind::In)
+      .addOperand("ComplexToRealWeights", OperandKind::In)
+      .addMember(MemberType::Int64, "WindowSize")
+      .addMember(MemberType::Int64, "WindowStride")
+      .addMember(MemberType::Boolean, "MagnitudeSquared")
+      .autoVerify(VerifyKind::NoVerify)
+      .autoIRGen();
+
+  BB.newInstr("MFCC")
+      .addOperand("Coefficients", OperandKind::Out)
+      .addOperand("Spectrogram", OperandKind::In)
+      .addOperand("MelWeights", OperandKind::In)
+      .addOperand("MelRanges", OperandKind::In)
+      .addOperand("DctMat", OperandKind::In)
+      .addMember(MemberType::Float, "SampleRate")
+      .addMember(MemberType::Float, "LowerFrequency")
+      .addMember(MemberType::Float, "UpperFrequency")
+      .addMember(MemberType::Int64, "FilterBankCount")
+      .addMember(MemberType::Int64, "NumCoefficients")
+      .autoVerify(VerifyKind::NoVerify)
```

Comment:
Is there no verification that can be done here on types/shapes/elem kinds?

Reply:
I put some verifications for element kinds.